src/pages/Vote/index.spec.tsx

```diff
@@ -1,19 +1,222 @@
 import React from 'react';
+import { act } from 'react-dom/test-utils';
+import { waitFor, fireEvent } from '@testing-library/react';
 import renderComponent from 'testUtils/renderComponent';
-import { useGetVestingVaults } from 'clients/api';
+import fakeAccountAddress, { altAddress } from '__mocks__/models/address';
+import transactionReceipt from '__mocks__/models/transactionReceipt';
+import { vaults } from '__mocks__/models/vaults';
+import proposals from '__mocks__/models/proposals';
+import { useGetVestingVaults, getCurrentVotes, setVoteDelegate, getProposals } from 'clients/api';
+import PATHS from 'constants/path';
+import useSuccessfulTransactionModal from 'hooks/useSuccessfulTransactionModal';
+import en from 'translation/translations/en.json';
+import BigNumber from 'bignumber.js';
+import { cloneDeep } from 'lodash';
+import TEST_IDS from './testIds';
 import Vote from '.';
 
 jest.mock('clients/api');
+jest.mock('hooks/useSuccessfulTransactionModal');
 
 describe('pages/Vote', () => {
   beforeEach(() => {
     (useGetVestingVaults as jest.Mock).mockImplementation(() => ({
       data: [],
       isLoading: false,
     }));
+    (getProposals as jest.Mock).mockImplementation(() => ({
+      proposals,
+      limit: 10,
+      total: 100,
+      offset: 10,
+    }));
+    (setVoteDelegate as jest.Mock).mockImplementation(() => transactionReceipt);
   });
 
   it('renders without crashing', async () => {
     renderComponent(<Vote />);
   });
+
+  it('opens create proposal modal when clicking text', async () => {
+    const { getByText } = renderComponent(<Vote />);
+    const createProposalButton = getByText(en.vote.createProposalPlus);
+
+    act(() => {
+      fireEvent.click(createProposalButton);
+    });
+
+    waitFor(() => getByText(en.vote.pages.proposalInformation));
+  });
+
+  it('opens delegate modal when clicking text with connect wallet button when unauthenticated', async () => {
+    const { getByText, getAllByText } = renderComponent(<Vote />);
+    const delgateVoteText = getByText('Delegate your voting power');
```

Comment:
Let's get this from the translation file

Reply:
The text in the translation file doesn't work because it includes markup

Reply:
I'd suggest using a test ID then